src/main/java/com/rapid7/client/dcerpc/io/PacketInput.java

```diff
@@ -68,6 +68,33 @@ public int readReferentID()
         return result;
     }
 
+    /*
```

Comment:
Need help here, perhaps readChars can be a part of readString (or one of its variants); but I'm not seeing it. Note that readChars helps me read the NetprPathCanonicalizeResponse, which is an `unsigned char* outBuf` according to https://msdn.microsoft.com/en-us/library/cc247258.aspx

Reply:
Ah, so this is not a string.  The conversion there might be in how you initialized the variables (a bit all over the place); doing the reads up front helps, you also don't need currentOffset (since that has meaning elsewhere).  Also why are you reading in shorts if this is a char*?

Reply:
See wireshark screencap:

![screen shot 2017-10-04 at 12 53 46 pm](https://user-images.githubusercontent.com/18122882/31188512-23af3c1a-a903-11e7-9e65-152ee7dfd047.png)

The NetPathCanonicalizeResponse returns
``` 
{ 
  int length; //this equals to the outBufLen in the request
  byte[length] path; //this is null terminated, with 4 bytes per char
  int pathType;
  int returnCode;
}
```
To decode the above path, there was no existing read method. readString is similar, but has expectations of offset, etc. Perhaps there is a more elegant way to express this readChars as a subset of readString.